server/index.js

```diff
@@ -1,14 +1,89 @@
 require('dotenv/config');
 const express = require('express');
+const pg = require('pg');
 const errorMiddleware = require('./error-middleware');
 const staticMiddleware = require('./static-middleware');
+const uploadsMiddleware = require('./uploads-middleware');
+const ClientError = require('./client-error');
 
-const app = express();
+const db = new pg.Pool({
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
 
+const app = express();
 app.use(staticMiddleware);
+const jsonMiddleware = express.json();
+app.use(jsonMiddleware);
 
-app.use(errorMiddleware);
+app.post('/api/users/', (req, res) => {
+  const {
+    username,
+    password,
+    avatarUrl,
+    email
+  } = req.body;
+  if (!username || !password || !avatarUrl || !email) {
+    res.status(400).json({
+      error: 'username, password, avatarUrl and email are required fields'
+    });
+    return;
+  }
+  const sql = `
+    insert into "users" ("username", "password", "avatarUrl", "email")
+    values ($1, $2, $3, $4)
+    returning *
+  `;
+  const params = [username, password, avatarUrl, email];
+  db.query(sql, params)
+    .then(result => {
+      const [user] = result.rows;
+      res.status(201).json(user);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'an unexpected error occurred'
+      });
+    });
```

Comment:
Pass the error to `next()` so that your error middleware can deal with it and you don't end up with this same block of code 10 times.